YAXLib/Exceptions/YAXException.cs

```diff
@@ -8,7 +8,7 @@ namespace YAXLib.Exceptions
     /// <summary>
     ///     The base for all exception classes of YAXLib
     /// </summary>
-    public class YAXException : Exception
+    public abstract class YAXException : Exception
```

Comment:
This is a breaking change, but maybe unlikely it's used.

I guess it's good practice to make the contructors now protected

Reply:
Doubting if we should fix this

Reply:
Would also qualify as unlikely to be used. Helps avoiding usage of a general exception class by contributors.